datafusion/core/tests/dataframe.rs

```diff
@@ -138,7 +138,7 @@ async fn sort_on_distinct_unprojected_columns() -> Result<()> {
         Arc::new(schema.clone()),
         vec![
             Arc::new(Int32Array::from_slice([1, 10, 10, 100])),
-            Arc::new(Int32Array::from_slice([2, 12, 12, 120])),
+            Arc::new(Int32Array::from_slice([2, 3, 4, 5])),
```

Comment:
This is also a newly added test -- and since the values of (a, b) are distinct where (a) was, it didn't show wrong results. If you change this data, prior to this PR it does show incorrect results

I fixed the sort to be on a valid column and added a test for incorrect columns